lib/explorer/data_frame.ex

```diff
@@ -4205,6 +4205,82 @@ defmodule Explorer.DataFrame do
     sample(df, 1.0, seed: opts[:seed], shuffle: true)
   end
 
+  @doc """
+  Transpose a DataFrame.
+
+  > #### Warning {: .warning}
+  >
+  > This is an expensive operation since data is stored in a columnar format.
+
+  ## Options
+
+    * `:header` - When a string is passed, name of the header column is set to the value. When `header` is `true`, name is set to `column` (default: `false`)
+    * `:columns` - names for non header columns. Length of column_names should match the row count of data frame. (default: `nil`)
+
+  ## Examples
+
+      iex> df = Explorer.DataFrame.new(a: ["d", nil], b: [1, 2], c: ["a", "b"])
+      iex> Explorer.DataFrame.transpose(df)
+      #Explorer.DataFrame<
+        Polars[3 x 2]
+        column_0 string ["d", "1", "a"]
+        column_1 string [nil, "2", "b"]
+      >
+
+      iex> df = Explorer.DataFrame.new(a: ["d", nil], b: [1, 2], c: ["a", "b"])
+      iex> Explorer.DataFrame.transpose(df, columns: ["x", "y"])
+      #Explorer.DataFrame<
+        Polars[3 x 2]
+        x string ["d", "1", "a"]
+        y string [nil, "2", "b"]
+      >
+
+      iex> df = Explorer.DataFrame.new(a: ["d", nil], b: [1, 2], c: ["a", "b"])
+      iex> Explorer.DataFrame.transpose(df, header: "name")
+      #Explorer.DataFrame<
+        Polars[3 x 3]
+        name string ["a", "b", "c"]
+        column_0 string ["d", "1", "a"]
+        column_1 string [nil, "2", "b"]
+      >
+  """
+  @doc type: :single
+  @spec transpose(df :: DataFrame.t(), opts :: Keyword.t()) :: DataFrame.t()
+  def transpose(df, opts \\ []) do
+    opts =
+      Keyword.validate!(opts, header: false, columns: nil)
+
+    header =
+      case opts[:header] do
+        false -> false
+        nil -> nil
```

Comment:
```suggestion
```

Reply:
To avoid `to_column_name` convert it to string, I added that clause -> `nil -> nil` ?

Reply:
When it would ever be nil? The accepted values should be Boolean or string, no?

Reply:
Atoms are also accepted- they are converted to strings by to_column(). So when user passes nil, it will be converted to "nil". Just to guard it I put that condition. Is nil to "nil" conversion is acceptable?

Reply:
Handling atoms makes sense. Would an `ArgumentError` be more appropriate for `header: nil`?

Also I think guards on the `to_column_name` clause might be helpful.

Reply:
If we disallow `nil`, user may pass `"nil"`.  `df[nil]` is valid, even though it looks funny. 
```elixir
iex> dft = DF.transpose(df, header: "nil", columns: ["x", "y"])
iex> dft[nil]
#Explorer.Series<
  Polars[3]
  string ["a", "b", "c"]
>
```
Not much we can do  here. I have accepted Jose change code suggestion and removed the statement.